dvc/stage/__init__.py

```diff
@@ -256,12 +256,7 @@ def changed_deps(self):
             return False
 
         if self.is_callback:
```

Comment:
The whole `callback` logic can be deleted now, it has been obsoleted since 1.0

Reply:
How so? The callback is still present on the outputs during run/repro, so let's remove it later at the same time.

Reply:
We have a checkbox https://github.com/iterative/dvc/issues/4841 . Sure, can do that separately, just saying.